sqlmesh/core/macros.py

```diff
@@ -1157,6 +1158,151 @@ def deduplicate(
     return query
 
 
+@macro()
+def date_spine(
+    evaluator: MacroEvaluator,
+    datepart: exp.Expression,
+    start_date: exp.Expression,
+    end_date: exp.Expression,
+) -> t.Union[exp.Alias, exp.Select]:
+    """Returns an aliased QUERY to build a date spine with the given datepart, and range of start_date and end_date. Useful for joining as a date lookup table.
+
+    Args:
+        datepart: The datepart to use for the date spine: day, week, month, year
+        start_date: The start date for the date spine in format YYYY-MM-DD
+        end_date: The end date for the date spine in format YYYY-MM-DD
+
+    Example:
+        >>> from sqlglot import parse_one
+        >>> from sqlglot.schema import MappingSchema
+        >>> from sqlmesh.core.macros import MacroEvaluator
+        >>> sql = "@date_spine('week', '2022-01-20', '2024-12-16')"
+        >>> MacroEvaluator(dialect='duckdb').transform(parse_one(sql)).sql()
+        'SELECT UNNEST(GENERATE_SERIES(CAST(\'2022-01-20\' AS DATE), CAST(\'2024-12-16\' AS DATE), (7 *INTERVAL \'1\' DAY))) AS "date_week"'
+
+    """
+    if datepart.name not in ("day", "week", "month", "year"):
```

Comment:
I'd consider assigning `datepart.name` to a variable to avoid invoking it repeatedly.

Reply:
done